Assorted Typing Fixups in pandas.io

The parquet, packers, html, feather_format, json.table_schema, json.normalize, and json.json could all be removed from the mypy.ini blacklist if we switch them over to absolute imports per the discussion in #25923 

Here's the complete list of errors:

```sh
pandas/io/parquet.py:8: error: Module 'pandas' has no attribute 'DataFrame'
pandas/io/packers.py:58: error: Module 'pandas' has no attribute 'Categorical'
pandas/io/packers.py:58: error: Module 'pandas' has no attribute 'CategoricalIndex'
pandas/io/packers.py:58: error: Module 'pandas' has no attribute 'DataFrame'
pandas/io/packers.py:58: error: Module 'pandas' has no attribute 'DatetimeIndex'
pandas/io/packers.py:58: error: Module 'pandas' has no attribute 'Float64Index'
pandas/io/packers.py:58: error: Module 'pandas' has no attribute 'Index'
pandas/io/packers.py:58: error: Module 'pandas' has no attribute 'Int64Index'
pandas/io/packers.py:58: error: Module 'pandas' has no attribute 'Interval'
pandas/io/packers.py:58: error: Module 'pandas' has no attribute 'IntervalIndex'
pandas/io/packers.py:58: error: Module 'pandas' has no attribute 'MultiIndex'
pandas/io/packers.py:58: error: Module 'pandas' has no attribute 'NaT'
pandas/io/packers.py:58: error: Module 'pandas' has no attribute 'Panel'
pandas/io/packers.py:58: error: Module 'pandas' has no attribute 'Period'
pandas/io/packers.py:58: error: Module 'pandas' has no attribute 'PeriodIndex'
pandas/io/packers.py:58: error: Module 'pandas' has no attribute 'RangeIndex'
pandas/io/packers.py:58: error: Module 'pandas' has no attribute 'Series'
pandas/io/packers.py:58: error: Module 'pandas' has no attribute 'TimedeltaIndex'
pandas/io/packers.py:58: error: Module 'pandas' has no attribute 'Timestamp'
pandas/io/html.py:17: error: Module 'pandas' has no attribute 'Series'
pandas/io/feather_format.py:7: error: Module 'pandas' has no attribute 'DataFrame'
pandas/io/feather_format.py:7: error: Module 'pandas' has no attribute 'Int64Index'
pandas/io/feather_format.py:7: error: Module 'pandas' has no attribute 'RangeIndex'
pandas/io/json/table_schema.py:15: error: Module 'pandas' has no attribute 'DataFrame'
pandas/io/json/normalize.py:11: error: Module 'pandas' has no attribute 'DataFrame'
pandas/io/json/json.py:14: error: Module 'pandas' has no attribute 'DataFrame'
pandas/io/json/json.py:14: error: Module 'pandas' has no attribute 'MultiIndex'
pandas/io/json/json.py:14: error: Module 'pandas' has no attribute 'Series'
pandas/io/json/json.py:14: error: Module 'pandas' has no attribute 'isna'
pandas/io/json/json.py:14: error: Module 'pandas' has no attribute 'to_datetime'
```

--- a/mypy.ini
+++ b/mypy.ini
@@ -140,27 +140,6 @@ ignore_errors=True
 [mypy-pandas.io.clipboards]
 ignore_errors=True
 
-[mypy-pandas.io.feather_format]
-ignore_errors=True
-
-[mypy-pandas.io.html]
-ignore_errors=True
-
-[mypy-pandas.io.json.json]
-ignore_errors=True
-
-[mypy-pandas.io.json.normalize]
-ignore_errors=True
-
-[mypy-pandas.io.json.table_schema]
-ignore_errors=True
-
-[mypy-pandas.io.packers]
-ignore_errors=True
-
-[mypy-pandas.io.parquet]
-ignore_errors=True
-
 [mypy-pandas.io.pytables]
 ignore_errors=True
 

